### ISSUE TITLE
Reworking the Foundation version (v0.3.1)

### ISSUE THREAD

**Issue by Gipson62** [atlas77-lang/Atlas77#61]
# Reworking the Foundation version ([v0.3.1](https://github.com/RedGear-Studio/Atlas77/releases/tag/v0.3.1)) to the Keystone version (v0.4.0)

## 1. Reworking the grammar
- [x] #80
## Implementing a Lexer and a Parser
- [x] #74
- [x] #75
- [x] #76
- #78
- #77
## Implement a basic type checker
- #64
> `string` will be a primitive types at first and will switch to a type in the std later as `String` same for `List`
- #73
- #79
## External call
- [x] #81
- #82
## Implement a basic standard library
- [x] #66
- [x] #67
- [x] #68
- [x] #69
- [x] #70
- [x] #71
- #72

> All the issues that don't have a checkbox anymore are issues that won't be coming in the 0.4 version

**Comment by Gipson62** [comment]
Unfortunately, the type checker won't make it into the 0.4 release. Implementing it requires significant rework, including building an entirely new Parser and, consequently, a new Runtime. Because of this, the type checker has been deferred to the 0.5 roadmap.

Additionally, structs and enums won't be part of 0.4 either. The current version already includes plenty of new features and improvements, and I believe it's in a solid state as it is.

For 0.4, the focus will now shift to Quality-of-Life (QoL) updates and bug fixes, if necessary. Once the standard library is complete, I'll consider 0.4 finalized.
